sdk/core/core/src/az_json_string.c

```diff
@@ -74,47 +80,56 @@ AZ_NODISCARD az_span az_json_esc_encode(az_result_byte c) {
     case '\t': {
       return AZ_SPAN_FROM_STR("\\t");
     }
-    default: {
-      return az_span_null();
-    }
+    default: { return az_span_null(); }
   }
 }
 
-AZ_NODISCARD az_result az_span_reader_read_json_string_char(az_span_reader * self, uint32_t * out) {
+AZ_NODISCARD az_result az_span_reader_read_json_string_char(az_span * self, uint32_t * out) {
```

Comment:
Ideally self would not be passed by *

Reply:
yeah, it was a little more refactoring to update to by-value.
I think that could be handled in a different effort (new PR)  since it is a chain of callings.  Same thing with the other ones below

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/348